R/RaMP.R

```diff
@@ -0,0 +1,252 @@
+#' @importFrom methods setClassUnion
+#' 
+#' @importClassesFrom DBI DBIDriver
+#'
+#' @noRd
+setClassUnion("DBIDriverOrNULL", c("DBIDriver", "NULL"))
+
+#' Maybe have an additional slot of type `list` with additional information/
+#' metadata retrieved from the database?
+#'
+#' @noRd
+setClass(
+    "RaMP",
+    slots = c(
+        driver = "DBIDriverOrNULL",
+        dbname = "character",
+        username = "character",
+        conpass = "character",
+        host = "character",
+        port = "integer"
+    ),
+    prototype = prototype(
+        driver = NULL,
+        dbname = character(),
+        username = character(),
+        conpass = character(),
+        host = character(),
+        port = integer()
+    ))
+
+#' Helper function to return the connection to the database, defined by the
+#' internal settings of the RaMP object.
+#'
+#' @param x `RaMP` object.
+#'
+#' @return a DB connection object.
+#'
+#' @noRd
+.dbcon <- function(x) {
+    con <- dbConnect(x@driver, dbname = .dbname(x), user = .username(x),
+                     password = .conpass(x), host = .host(x), port = .port(x))
+}
+
+.dbname <- function(x) {
+    if (length(x@dbname)) x@dbname
+    else NULL
+}
+
+.username <- function(x) {
+    if (length(x@username)) x@username
+    else NULL
+}
+
+.conpass <- function(x) {
+    if (length(x@conpass)) x@conpass
+    else NULL
+}
+
+.host <- function(x) {
+    if (length(x@host)) x@host
+    else NULL
+}
+
+.port <- function(x) {
+    if (length(x@port)) x@port
+    else NULL
+}
+
+#' Helper function to check if the connection is/will be to a
+#' SQLite database
+#'
+#' @noRd
+.is_sqlite <- function(x) {
+    inherits(x@driver, "SQLiteDriver")
+}
+
+#' @importMethodsFrom methods show
+#'
+#' @importFrom DBI dbDisconnect
+#' 
+#' @exportMethod show
+#'
+#' @rdname RaMP
+setMethod("show", "RaMP", function(object) {
+    if (is.null(object@driver))
+        cat("Empty RaMP object")
+    con <- .dbcon(object)
+    on.exit(dbDisconnect(con))
+    cat(class(object), "\n")
+    ## Maybe get some additional information from the database with e.g.
```

Comment:
would be nice to eventually also show here some information on the version of the used resource and some other statistics like number of analytes or similar. I was however not sure how that info could be best extracted from the database.

Reply:
Hi Johannes, Thanks for the explanations. It all looks good to me. 
For entity information you can use "select * from entity_status_info", but you might want to group on status_category and sum entities from each data source. The 'db_version' table is supposed to hold current and previous DB version information, but the processes to generate a new data table I think at some point truncated the table so while it holds current information, the 'history' aspect hasn't been well maintained, although that was my intention.

Everything is looking good to me. As I mentioned in the email, everyone here is on-board with the changes you proposed.